### PR TITLE
done ticket 7

### PR DESCRIPTION
**Pull request by ThePotato97**

### CODE REVIEW

File: api/__tests__/app.test.js [hunk 1]
```diff
@@ -130,6 +130,67 @@ describe("/api", () => {
...
   });
+  describe("POST /api/articles/:article_id/comments", () => {
+    it("should return the posted comment", () => {
+      return request(app)
+        .post("/api/articles/1/comments")
+        .send({
+          username: "butter_bridge",
+          body: "Hello World!",
+        })
+        .expect(200)
```
Comment (pippaamy) [inline api/__tests__/app.test.js]: this should expect a 201

File: api/__tests__/app.test.js [hunk 2]
```diff
@@ -130,6 +130,67 @@ describe("/api", () => {
...
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            votes: expect.any(Number),
+            created_at: expect.any(String),
+            author: "butter_bridge",
+            body: "Hello World!",
+            article_id: 1,
+          });
+        });
+    });
```
Comment (pippaamy) [inline api/__tests__/app.test.js:174]: can we add a 201 test that it ignores unnecessary properties put on the posted object

File: api/__tests__/app.test.js [hunk 3]
```diff
@@ -130,6 +130,67 @@ describe("/api", () => {
...
+          username: "butter_bridge",
+          body: "Hello World!",
+        })
+        .expect(400)
+        .then((res) => {
+          const { body } = res;
+          expect(body.msg).toBe("Bad Request");
+        });
+    });
+    it("should return 400 if the article id invalid", () => {
```
Comment (pippaamy) [inline api/__tests__/app.test.js]: this description should be 404

File: api/__tests__/app.test.js [hunk 4]
```diff
@@ -130,6 +130,67 @@ describe("/api", () => {
...
+          username: "butter_bridge",
+          body: "Hello World!",
+        })
+        .expect(404)
+        .then((res) => {
+          const { body } = res;
+          expect(body.msg).toBe("Article ID does not exist");
+        });
+    });
+    it("should return an error when the article id doesn't exist", () => {
```
Comment (pippaamy) [inline api/__tests__/app.test.js]: you've done the same test twice

File: api/__tests__/app.test.js [hunk 5]
```diff
@@ -130,6 +130,67 @@ describe("/api", () => {
...
+    });
+    it("should be able to handle an invalid article id", () => {
+      return request(app)
+        .post("/api/articles/test/comments")
+        .send({
+          username: "butter_bridge",
+          body: "Hello World!",
+        })
+        .expect(400)
+        .then((res) => {
```
Comment (pippaamy) [inline api/__tests__/app.test.js:183]: we need a test for username doesnt exist
and a test for missing fields - eg no body

File: api/controllers/articles.controller.js [hunk 6]
```diff
@@ -29,6 +26,16 @@ exports.getCommentsByArticleId = (req, res, next) => {
     .catch((err) => next(err));
 };
 
+exports.postComment = (req, res, next) => {
+  const { username, body } = req.body;
+  const { article_id } = req.params;
+  addComment(article_id, username, body)
+    .then((comment) => {
+      res.status(200).send({ comment });
```
Comment (pippaamy) [inline api/controllers/articles.controller.js]: 201 expected here

File: api/models/articles.model.js [hunk 7]
```diff
@@ -20,16 +29,24 @@ exports.selectArticles = () => {
...
+      return db.query(
+        `
+    INSERT INTO comments(article_id, author, body)
+    VALUES ($1, $2, $3)
+    RETURNING *;
+    `,
+        [id, username, body]
+      );
+    })
+    .then(({ rows }) => rows.pop());
```
Comment (pippaamy) [inline api/models/articles.model.js]: pop will mutate the array, lets think of a different way of doing this

**Review by pippaamy — COMMENTED** [review]
good job, lets make those changes and pop in another pr :)

**Review by xViolaine — APPROVED** [review]

**Pull request merged**